lib/matplotlib/markers.py

```diff
@@ -265,7 +285,7 @@ def _recache(self):
         self._alt_transform = None
         self._snap_threshold = None
         self._joinstyle = JoinStyle.round
```

Comment:
Why not do this 
```suggestion
        self._joinstyle = self._user_joinstyle or JoinStyle.round
```
Why not do this, and then leave `get_joinstyle` and `get_capstyle` as they are below? (ie. just returning `self._capstyle`)

Reply:
Good thinking, we cannot set `self._joinstyle` here, because it gets overwritten with some `_marker_function()`s. However, I have simplified the `get_capstyle()` and `get_joinstyle`. And I have also added tests for these.